### DISCUSSION TITLE
Particles

### DISCUSSION THREAD

**Discussion by f-t-alves** [kaisermann/kaisermann#27] [Ideas]
Discussão dos próximos passos e features desejadas

# Features

- Objeto célula
  - Tipo
    - Cor
    - Física
    - Química
  - Velocidade
- Grid de células
- Física
  - Contato
  - Aceleração da gravidade
  - Densidade e empuxo
  - Arraste (velocidade máxima fixa ou força variável)
  - Difusão ('espalhamento')
- Química
  - Fogo
  - Solubilidade
  - Ácido

# Próximos passos

- Refatorar para múltiplos arquivos
- Implementar células

**Comment by f-t-alves** [comment]
O grid tô em dúvida se é melhor fazer um listão com lógica de endereço ou uma lista de listas mesmo. Se a lista careta de JS por si só já for uma lista encadeada ao invés de um vetor contíguo em memória, grandes chances de lista de listas ser mais performático.

**Reply by kaisermann** [reply]
Vamo simplificar nesse começo 

```ts 
type Cell = {
  row: number
  column: number
  // ...
}

type Grid = {
  cells: Array<Cell>
}
```

**Reply by f-t-alves** [reply]
É possível botar funções dentro do objeto? Por exemplo:
```TS
type Grid = {
  nRow: 20,
  nCol: 20,

  cells: Array<Cell>,

  function getCellbyGridCoords(i, j){
    let index = (nRow - 1) * j + i;
    return this.cells[index];
  }
}
```
